### PR TITLE
Allow unspecified values to be optional

### LINKED ISSUE #41: Enable unspecified parameters values to functions to be optional under certain circumstances
See discussion at #40

### PR DESCRIPTION
**Pull request by Alxandr**
Fixes #41

---

This change is [Reviewable](https://reviewable.io/reviews/yolodev/babel-dts-generator/42)

### CODE REVIEW

**Comment by Alxandr** [comment]
Review status: 0 of 13 files reviewed at latest revision, 1 unresolved discussion.

---

_[src/generators/ast.js, line 359 [r1]](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42#-KGZ2eNMDS9kafgFm_wp:-KGZ2eNMDS9kafgFm_wq:-1513883598) ([raw file](https://github.com/yolodev/babel-dts-generator/blob/ab3e7f4f3dcfd1cb364513fe47f065d88965b02b/src/generators/ast.js#L359)):_
Don't leave old code as comments -.-

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42)_

**Comment by cmichaelgraham** [comment]
I looked through the changes.  I like the way this works.  I need to do some testing this weekend (Aurelia is experiencing some duplicate comments with the plugin & preset combo they use).  I can validate the existing behaviors as part of that task.

Thanks for the continued investment in this plugin.

---

Reviewed 13 of 13 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42#-:-KGZU7VJdXeifgo1z7cu:-1730487414)_

**Comment by Alxandr** [comment]
np. I still need to fix that code comment I left. It was just too late yesterday :P. Will see to it today.

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42#-:-KGaok2ZdlvPlftiaGwk:957203429)_

**Comment by doktordirk** [comment]
class parameters are now typed as `null`and not as `any`anymore

File: test/debug.js [hunk 1]
```diff
@@ -24,12 +24,11 @@ const result = transform(content, {
   //loose: 'all',
   plugins: [
     'syntax-flow',
-    ['./lib/index', {
+    ['babel-dts-generator', {
```
Comment (doktordirk) [inline test/debug.js:27]: that's intended?

**Comment by Alxandr** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

_[test/debug.js, line 27 [r1]](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42#-KGZ1bnt94_q3k8657z2-r1-27:-KGiEqUH-BfDvkN8_b0U:-1626298895) ([raw file](https://github.com/yolodev/babel-dts-generator/blob/ab3e7f4f3dcfd1cb364513fe47f065d88965b02b/test/debug.js#L27)):_
Yeah. Should have been change earlier really. At some point babel stopped working with local paths as plugins. Might be working again, but figured it's just better to use it like it would be on a actual project.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42)_

**Comment by Alxandr** [comment]
Please provide failing input along with what you would expect as output.

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42#-:-KGiFFxVlyTrkvIbrcep:-171783309)_

**Comment by Alxandr** [comment]
Review status: 12 of 13 files reviewed at latest revision, 1 unresolved discussion.

---

_[src/generators/ast.js, line 359 [r1]](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42#-KGZ2eNMDS9kafgFm_wp:-KGiGh4ZuMNpaELoBCTE:-499146009) ([raw file](https://github.com/yolodev/babel-dts-generator/blob/ab3e7f4f3dcfd1cb364513fe47f065d88965b02b/src/generators/ast.js#L359)):_
Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42)_

**Comment by doktordirk** [comment]
class/property.src.js

``` js
export class Foo {
  foo: number = 5;
  get bar(): number {
    return 10;
  }
  baz;
}
```

class/property.expected.ts

``` js
declare module 'spec' {
  export class Foo {
    foo: number;
    bar: number;
    baz: any;
  }
}
```

currently yields baz: null; but should be/was 'any'

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42#-:-KGkPKKcmiMZhXPsr638:-1702577988)_

**Comment by Alxandr** [comment]
Done. Any more specs you want added?

---

Review status: 12 of 17 files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42#-:-KGkaDP_Mzya2uPraYWt:-1112159896)_

**Comment by doktordirk** [comment]
not at this point. works perfect now. 

---

Review status: 12 of 17 files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42#-:-KGkmPHjhAjYRrNwYI_O:821730076)_

**Comment by doktordirk** [comment]
Reviewed 2 of 5 files at r3.
Review status: 14 of 17 files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42)_

**Comment by doktordirk** [comment]
Reviewed 2 of 5 files at r3.
Review status: 16 of 17 files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42)_

**Comment by Alxandr** [comment]
Reviewed 11 of 13 files at r1, 5 of 5 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_[test/index.js, line 124 [r3]](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42#-KGlGJZ7_HiKN32y7pHz:-KGlGJZ9QCUkBYxkMLhs:1607712484) ([raw file](https://github.com/yolodev/babel-dts-generator/blob/8d9768fd7428fd5be62988c3c63e018284a7e80e/test/index.js#L124)):_
I don't know why `trimEnd` works, and can't find docs on it, which is somewhat scary. According to MDN `trimRight` do work though, and I've tested this in node. Will have to change it.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42)_

**Comment by Alxandr** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_[test/index.js, line 124 [r3]](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42#-KGlGJZ7_HiKN32y7pHz:-KGlI6adGAjWNeHMTZv2:-499146009) ([raw file](https://github.com/yolodev/babel-dts-generator/blob/8d9768fd7428fd5be62988c3c63e018284a7e80e/test/index.js#L124)):_
Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42)_

**Comment by Alxandr** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

_[a discussion](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42#-KGlIxFHxEnGAq2ILmFD:-KGlIxFHxEnGAq2ILmFE:-2041613946) (no related file):_
I think this is ready. @cmichaelgraham will you look over the last changes to see if they're ok? Especially the spec changes, as these might impact Aurelia.

@doktordirk You're obviously more than welcome to look over things again.

+review 

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42)_

**Comment by doktordirk** [comment]
Reviewed 3 of 13 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_[a discussion](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42#-KGlIxFHxEnGAq2ILmFD:-KGlOwbu8pwBcqRiInq0:-2076173905) (no related file):_
i really only can/want to review the specs and the final outcome for our plugins. and those are fine :+1: 

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42)_

**Comment by cmichaelgraham** [comment]
Reviewed 5 of 5 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42)_

**Comment by cmichaelgraham** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_[a discussion](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42#-KGlIxFHxEnGAq2ILmFD:-KGwFe6Jvd_9am0iEFD8:-510466997) (no related file):_
:+1:

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42)_

**Comment by cmichaelgraham** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42#-:-KGwGI6amOkhJVt8GMr8:1417088721)_

**Comment by cmichaelgraham** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/42)_
